tests/PHPStan/Rules/Comparison/data/bug-5474.php

```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types = 1);
+
+namespace Bug5474;
+
+$data = ['test' => 1];
+$data2 = ['test' => 1];
+$data3 = ['test' => 5];
+
+/**
+ * @phpstan-param array{test: int} $data
+ * @return void
+ */
+function testData(array $data) {
+	var_dump($data);
+}
+
+/**
+ * @phpstan-return array{test: int}
+ */
+function returnData3() {
+	return ['test' => 5];
+}
+
+if ($data !== $data2) {
```

Comment:
Same reason as above - you can put this code including the variables above into another function (`function test(): void`)